app/views/step_by_step_pages/_nav.html.erb

```diff
@@ -43,6 +43,7 @@
     }
   ]
   tabs.delete_at(1) unless @step_by_step_page.can_be_edited?
+  tabs[0][:label] = "View steps" unless @step_by_step_page.can_be_edited?
```

Comment:
I'd move this to line 11.

```ruby
label: @step_by_step_page.can_be_edited? ? "Edit steps" : "View steps",
```

Means the decision happens in one place, and reduces chance of a bug creeping in (e.g. if we reorder the tabs in future, your current code might accidentally replace the wrong text).